agent/tool-scripts/postprocess/pidstat-postprocess

```diff
@@ -149,35 +162,36 @@ while ($line = <PIDSTAT_TXT>) {
 		$pidstat{memory_usage}{virtual_size}{$pid}{$timestamp_ms} = $vsz;
 		$pidstat{memory_usage}{resident_set_size}{$pid}{$timestamp_ms} = $rss;
 		$timestamps{$timestamp_ms}++;
-	}
-}
-close(PIDSTAT_TXT);
 
-# fill in any missing data with zeros
-my $htmlpage;
-my $graph;
-my $pid;
-my $timestamp_ms;
-foreach  $htmlpage ( keys %pidstat ) {
-	foreach  $graph ( keys %{ $pidstat{$htmlpage} } ) {
-		foreach $pid ( keys %{ $pidstat{$htmlpage}{$graph} } ) {
-			foreach $timestamp_ms ( sort {$a <=> $b} (keys %timestamps ) ) {
-				if (! defined($pidstat{$htmlpage}{$graph}{$pid}{$timestamp_ms})) {
-					$pidstat{$htmlpage}{$graph}{$pid}{$timestamp_ms} = 0;
-				}
-			}
-		}
+    # parse data every 10,000 lines.
+    if ($lines_scanned % $lines_limit == 0) {
+
+        # TODO: this code shold be multithreaded in order to speed up the process.
```

Comment:
Can you move this comment to an issue instead?

Reply:
sure